app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalCategoryController.php

```diff
@@ -74,13 +76,22 @@ public function show(KnowledgeBaseCategory $category): JsonResponse
             ]),
             'articles' => KnowledgeBaseArticleData::collection(
                 $category->knowledgeBaseItems()
+                    ->when(Feature::active('tags'), fn (Builder $query) => $query->with('tags'))
                     ->public()
                     ->get()
                     ->map(function ($item) {
                         return [
                             'id' => $item->getKey(),
                             'categoryId' => $item->category_id,
                             'name' => $item->title,
+                            'tags' => Feature::active('tags') ? $item->tags()
+                                ->orderBy('name')
+                                ->get()
+                                ->select([
+                                    'id',
+                                    'name',
+                                ])
```

Comment:
I think if you put the `select` before the `get` here it will make sure to actually scope the query to the DB, reducing the data in SQL.